index.js

```diff
@@ -196,7 +196,13 @@ Polyglot.prototype.t = function (key, options) {
   }
   if (typeof phrase === 'string') {
     opts = assign({}, opts);
-    result = choosePluralForm(phrase, this.currentLocale, opts.smart_count);
+    if (typeof opts.smart_count === 'object' && isValidSmartObject(opts.smart_count)) {
```

Comment:
The logic behind keeping that test out of the method was not wanting to invoke the method for every single translation invocation. The performance margin should be insignificant though so I will move it inside as you suggested.